Update moltres_xs.py to be compatible with OpenMC 0.13.2

smpark7 opened the pull request:
Fixes #205 

The following changes from OpenMC versions 0.12.2 to 0.13.2 affected `moltres_xs.py`:
- Renaming the `groups` parameter for `MGXS` group constant classes to `energy_groups`
  - openmc-dev/openmc#2147
- Removing the `legendre` filter in the diffusion coefficient group constants
  - openmc-dev/openmc#1909

smpark7 commented:
I just realized that this breaks backward compatibility with pre-0.13.2 OpenMC-generated files.

munkm commented:
> I just realized that this breaks backward compatibility with pre-0.13.2 OpenMC-generated files.

I was going to ask that! Ok, so maybe it would be helpful to add a warning or helpful error to users trying to load files that are that old? What do you think?

munkm commented on tutorial/godiva/godiva_openmc_1200.py:
```diff
@@ -21,7 +21,7 @@
 
 settings = openmc.Settings()
-settings.batches = 200
+settings.batches = 100
```
Is this change to save time?

smpark7 replied:
I'm not sure why the simulation at 900K used 100 batches while 1200K used 200 batches. It seems like a mistake so I wanted to standardize both to 100 batches.

munkm commented on python/moltres_xs.py:
```diff
@@ -300,10 +299,10 @@ def generate_openmc_tallies_xml(
             )
             domain_dict[id]["chi"] = mgxs.Chi(
-                domain=domain, groups=groups, name=str(id) + "_chi"
+                domain=domain, energy_groups=groups, name=str(id) + "_chi"
```
I think because we're switching the kwarg it also means we aren't supporting older openmc versions.

munkm commented:
I think it's ok to drop support for older versions of openmc. We should update the documentation for moltres to reflect this requirement though.

smpark7 commented:
This is ready for another review @munkm 

I made the following changes:
- pep8 format edits to `moltres_xs.py`
- added checks whether OpenMC version is 0.13.2
- updated the readme to state that only OpenMC v0.13.2 and later are supported
- updated the pytest files which use the same files as the tutorial

munkm reviewed (approved):
This looks good to me! Thanks for clarifying that the .h5 files and the godiva.json were updated using the new openmc version.

smpark7 commented:
Thank you for reviewing this PR!
